mage_ai/frontend/components/PipelineDetail/UploadPipeline/index.tsx

```diff
@@ -0,0 +1,175 @@
+          width={UNIT * 50}
+    />;
+      }, {
+  }, [
+    uploadedPipeline,
+  ], {
+    background: true,
+    uuid: 'confirm_upload',
+  });
+
+  const tableMemo = useMemo(() => {
+    const rows = [];
+    sortByKey(Object.entries(fileUploadProgress), ([name, _]) => name).forEach(([
+      filename,
+      progress,
+    ]) => {
+      const file: ApiErrorType & FileType = uploadedPipeline[filename];
+      const errorMessage = file?.message;
+
+      rows.push([
+        <div key={`name-${filename}`}>
+          <Text
+            overflowWrap
+            preWrap
+          >
+            {filename}
+          </Text>
+          {errorMessage && (
+            <Spacing mt={1}>
+              <Text danger small>
+                {errorMessage}
+              </Text>
+            </Spacing>
+          )}
+        </div>,
+        <ProgressBar
+          danger={!!errorMessage}
+          key={`progress-${filename}`}
+          progress={progress * 100}
+        />,
+      ]);
+    });
+
+    return (
+      <Table
+        columnFlex={[1, 4]}
+        columns={[
+          {
+            uuid: 'Filename',
+          },
+          {
+            uuid: 'Upload progress',
+          },
+        ]}
+        rows={rows}
+        uuid="block-runs"
+      />
+    );
+  }, [fileUploadProgress, uploadedPipeline]);
+
+  return (
+    <Panel
+      footer={(
+        <FlexContainer fullWidth justifyContent="space-between">
+          <Button onClick={() => onCancel()}>
+            Close
+          </Button>
+          <Spacing ml={1}>
+            <FlexContainer alignItems="center" justifyContent="flex-end">
+              <Text>Overwrite pipeline files</Text>
+              <Spacing ml={1}/>
+              <ToggleSwitch
+                checked={overwriteVariable}
+                onCheck={setOverwriteVariable}
+              />
+            </FlexContainer>
+          </Spacing>
+        </FlexContainer>
+      )}
+      headerTitle="Import pipeline from zip"
+    >
+      {hasFiles && (
+        <TableStyle>
+          {tableMemo}
+        </TableStyle>
+      )}
+
+      {!hasFiles && (
+        <FileUploader
+          directoryPath={null} // no directory needs to be specified in case of pipeline imports
+          overwrite = {overwriteVariable}
+          pipelineZip = {true}
+          setUploadedFiles={(uploadedFiles) => {
+            // @ts-ignore
+            setUploadedPipeline(uploadedFiles);
+            showPopupMenu();
+            fetchPipelines?.();
+          }}
+          // @ts-ignore
+          onDragActiveChange={setIsDragActive}
+          // @ts-ignore
+          setFileUploadProgress={setFileUploadProgress}
+        >
+          <DropZoneStyle>
+            <Text center>
+              {isDragActive && 'Drop pipeline zip to import' }
+              {!isDragActive && 'Click or drop pipeline zip to upload' }
+            </Text>
+          </DropZoneStyle>
```

Comment:
Please add a brief explanation of what files are required or allowed in the pipeline zip file (e.g. `metadata.yaml` is required; any Mage block files are allowed, etc). Without it, it seems a bit unclear on what exactly a user is allowed to upload for a "valid" pipeline.

You can use a `Text` component with the `warning` prop (to make the font color yellow) and some top padding (you can use the `Spacing` component). It might look something like this:
![image](https://github.com/mage-ai/mage-ai/assets/78053898/33791f28-068c-49da-b87e-4c44115ea088)

Reply:
![image](https://github.com/mage-ai/mage-ai/assets/33767023/8fb70c64-f5c7-40a1-8333-be22e501fde5)
Would this work?

Reply:
Instead of adding the text inside the footer of the Panel component, you can add it as a child component to FileUploader right under `DropZoneStyle` (where I made this PR comment).

Adding the following code block looks like this (adding the line break prevents the popup from stretching its width):
```
<Spacing mt={2}>
  <Text warning>
    The zip file should include the pipeline’s metadata.yaml file and each 
    <br />
    block’s script file in the root folder.
  </Text>
</Spacing>
```
![image](https://github.com/mage-ai/mage-ai/assets/78053898/3de2f248-396d-4c66-ba95-dd60db15079a)